Makefile

```diff
@@ -21,21 +22,38 @@ endif
 all:
 	$(MSBUILD)
 
-prepare::
+prepare:: prepare-props
 	git submodule update --init --recursive
 	nuget restore
 	(cd external/Java.Interop && nuget restore)
+
+prepare-props:
 	cp Configuration.Java.Interop.Override.props external/Java.Interop/Configuration.Override.props
+	./build-tools/scripts/generate-os-info Configuration.OperatingSystem.props
```

Comment:
Should this be `generate-os-info.sh` (`.sh` suffix)?

Should `generate-os-info` always write to stdout, and callers redirect to the intended output file? This would allow running it from the command-line for test purposes w/o accidentally nuking some file:

    $ ./build-tools/scripts/generate-os-info.sh
    ...output here...